src/pal/src/include/pal/stackstring.hpp

```diff
@@ -23,17 +23,25 @@ class StackString
 
     void DeleteBuffer()
     {
-        if (m_innerBuffer != m_buffer)
+        if ( m_innerBuffer != m_buffer)
             PAL_free(m_buffer);
 
         m_buffer = NULL;
         return;
     }
 
-    void ReallocateBuffer(SIZE_T count)
+    void ReallocateBuffer(SIZE_T count, BOOL preserve)
     {
         // count is always > STACKCOUNT here.
-        T * newBuffer = (T *)PAL_malloc((count + 1) * sizeof(T));
+        // We got so far, we will allocate a little extra
+        // to prevent frequent allocations
+#if _DEBUG
+        SIZE_T count_allocated = count;
```

Comment:
Reserve is always a best effort. What if this data structure is passed to another function and it wants to modify it, are we okay with paying the penalty of reallocating a new buffer if we are short of few characters ?